reader-osm/src/test/java/com/graphhopper/GraphHopperTest.java

```diff
@@ -887,7 +887,7 @@ public void testSRTMWithInstructions() {
                 setAlgorithm(ASTAR).setProfile(profile));
 
         ResponsePath res = rsp.getBest();
-        assertEquals(1625.4, res.getDistance(), .1);
+        assertEquals(1614.3, res.getDistance(), .1);
```

Comment:
This is the only test where my change lead to a distance different above `1m`, is this ok?

Reply:
I had a look at this and the difference of about `11m` is the result of multiple edges that have a different distance due to the change in this PR. The biggest contribution is an edge with a `6.9m` difference and the full geometry is shown here:

![image](https://user-images.githubusercontent.com/17603532/103231849-22c17780-4939-11eb-88ea-c5c6811b38e9.png)

Due to the simplification it looks like this:

![image](https://user-images.githubusercontent.com/17603532/103231904-4be20800-4939-11eb-85f2-c4db8d85761a.png)

Because point 2 is removed we get a longer less steep segment instead of a short steep and a long flat one and this leads to the different distance with and without simplification. Note that the long edge sampling that would normally prevent the loss of these elevation details is not enabled in this test.

So yes this PR changes the distance by a significant amount, but no this is nothing to worry about, because all we do now is that we store the distance of the geometry we store for this edge instead of storing a distance for a geometry that we do not keep anyway.